content/3_Silver/Graph_Traversal.mdx

```diff
@@ -1210,46 +1255,38 @@ public class BuildingTeams {
 <PySection>
 
 ```py
+import sys
 from collections import deque
 
-n, m = map(int, input().split())
+input = sys.stdin.readline
 
+n, m = map(int, input().strip().split())
 adj = [[] for _ in range(n)]
+team = [0] * n  # 0 -> has not been assigned a team yet
+
 for _ in range(m):
-	a, b = map(int, input().split())
+	a, b = map(int, map(int, input().strip().split()))
 	adj[a - 1].append(b - 1)
 	adj[b - 1].append(a - 1)
 
-assigned = [0 for _ in range(n)]
-valid = True
```

Comment:
just now realizing that this is replacing an old sol
uneasy about this bc the old one was consistent w/ java and c++, while this one isn't

Reply:
> _Note: Updated/changed the pre-existing Python BFS solution as I had a simpler, faster solution._

Yeah, I mentioned it in the description. If that's a concern, we just just use the old one. If so, I can try to see if there are any optimizations that can be made while still keeping it consistent with the other language's solutions (unless it's exactly the same).

Reply:
> I can try to see if there are any optimizations that can be made while still keeping it consistent

i'd love this